api/kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/HostPort.java

```diff
@@ -25,6 +25,8 @@ public final class HostPort {
 
     /**
      * Creates a host port.
+     * @param host the host - symbolic hostnames, FQDNs, IPv4, and IPv6 forms are supported
+     * @param port port number
      */
     public HostPort(String host, int port) {
         Objects.requireNonNull(host, "host cannot be null");
```

Comment:
I know not directly related to this PR, but I thought we were using a `null` hostname to convey bind to any interface so this NotNull restriction seems a little incongruous.

Reply:
I'm going to suggest we agree to disagree on this. I find it strange that we use one null semantic in one context and a different one in different contexts for what is ostensibly the same thing.  

One thought would be to add the validation in a factory method so that we can add some semantic notion around the different contexts but then consumers would still need to deal with different nullability expectations.